docs/how_to_guide/plot_05_sklearn_pipeline_cv_demo.md

```diff
@@ -0,0 +1,560 @@
+</figure>
+
+A pipeline is a sequence of data transformations leading up to a model. Each step before the final one transforms the input data into a different representation, and then the final model step fits, predicts, or scores based on the previous step's output and some observations. Setting up such machinery can be simplified using the [`Pipeline`](https://scikit-learn.org/1.5/modules/generated/sklearn.pipeline.Pipeline.html) class from scikit-learn.
+
+To set up a scikit-learn [`Pipeline`](https://scikit-learn.org/1.5/modules/generated/sklearn.pipeline.Pipeline.html), ensure that:
+
+1. Each intermediate step is a [scikit-learn transformer object](https://scikit-learn.org/stable/data_transforms.html) with a `transform` and/or `fit_transform` method.
+2. The final step is an [estimator object](https://scikit-learn.org/stable/developers/develop.html#estimators) with a `fit` method, or a model with `fit`, `predict`, and `score` methods.
+
+Each transformation step takes a 2D array `X` of shape `(num_samples, num_original_features)` as input and outputs another 2D array of shape `(num_samples, num_transformed_features)`. The final step takes a pair `(X, y)`, where `X` is as before, and `y` is a 1D array of shape `(n_samples,)` containing the observations to be modeled.
+
+You can define a pipeline as follows:
+```python
+from sklearn.pipeline import Pipeline
+
+# Assume transformer_i/predictor is a transformer/model object
+pipe = Pipeline(
+    [
+        ("label_1", transformer_1), 
+        ("label_2", transformer_2),
+        ...,
+        ("label_n", transformer_n),
+        ("label_model", model)
+    ]
+)
+```
+
+Note that you have to assign a label to each step of the pipeline.
+:::{tip}
+Here we used a placeholder `"label_i"` for demonstration; you should choose a more descriptive name depending on the type of transformation step.
+:::
+
+Calling `pipe.fit(X, y)` will perform the following computations:
+```python
+# Chain of transformations
+X1 = transformer_1.fit_transform(X)
+X2 = transformer_2.fit_transform(X1)
+# ...
+Xn = transformer_n.fit_transform(Xn_1)
+
+# Fit step
+model.fit(Xn, y)
+```
+And the same holds for `pipe.score` and `pipe.predict`.
+
+## Why pipelines are useful
+
+Pipelines not only streamline and simplify your code but also offer several other advantages. The real power of pipelines becomes evident when combined with the scikit-learn [`model_selection`](https://scikit-learn.org/1.5/api/sklearn.model_selection.html) module, which includes cross-validation and similar methods. This combination allows you to tune hyperparameters at each step of the pipeline in a straightforward manner.
+
+In the following sections, we will showcase this approach with a concrete example: selecting the appropriate basis type and number of bases for a GLM regression in NeMoS.
+
+## Combining basis transformations and GLM in a pipeline
+Let's start by creating some toy data.
+
+
+```{code-cell} ipython3
```

Comment:
This prints a UserWarning for using plotting functions in `_documentation_utils`. Suppress?